### ISSUE TITLE
`librz/cons/*` (TUI library/) - remove global state

### ISSUE THREAD

**Issue by ghost** [rizinorg/rizin#276]
# Remove global variables

## `librz/cons/input.c`
```c
#define I rz_cons_singleton()
```

**Additional context**

Also related: https://github.com/rizinorg/rizin/issues/985

**Comment by XVilka** [comment]
I don't think we need to use any external library. So far, Rizin's own is battle tested across different systems and environments, and much smaller than alternatives. We don't require all features of the "bigger" libraries, thus I think it's better to keep it.

**Comment by ret2libc** [comment]
That said, it would definitely benefit from more unit tests and better API, as all the other librz modules :(

**Comment by XVilka** [comment]
A good starting point would be removing global variables, e.g. in `librz/cons/editor.c`:
```c
/* TODO: remove global vars */
static char *lines = NULL;
static char *path = NULL;
static char prompt[32];
static int bytes, nlines, _n = 0;
```
